pkg/apis/machine/v1beta1/machine_webhook_test.go

```diff
@@ -1995,6 +2060,39 @@ func TestDefaultGCPProviderSpec(t *testing.T) {
 			expectedOk:    true,
 			expectedError: "",
 		},
+		{
+			testCase: "sets OnHostMaintance to TERMINATE if instance has guestAccelerators",
+			providerSpec: &gcp.GCPMachineProviderSpec{
+				GuestAccelerators: []*gcp.GCPAcceleratorConfig{
+					{
+						AcceleratorType:  "type",
+						AcceleratorCount: defaultGCPAcceleratorCount,
+					},
+				},
+			},
+			modifyDefault: func(p *gcp.GCPMachineProviderSpec) {
+				p.GuestAccelerators = []*gcp.GCPAcceleratorConfig{
+					{
+						AcceleratorType:  "type",
+						AcceleratorCount: defaultGCPAcceleratorCount,
+					},
+				}
+				p.OnHostMaintenance = "TERMINATE"
+			},
+			expectedOk:    true,
+			expectedError: "",
+		},
+		{
+			testCase: "sets OnHostMaintance to TERMINATE if instance si A2 machine type",
+			providerSpec: &gcp.GCPMachineProviderSpec{
+				MachineType: "a2-highgpu-1g",
+			},
+			modifyDefault: func(p *gcp.GCPMachineProviderSpec) {
+				p.MachineType = "a2-highgpu-1g"
+				p.OnHostMaintenance = "TERMINATE"
+			}, expectedOk: true,
```

Comment:
minor nit, just to make it consistent
```suggestion
			}, 
			expectedOk: true,
```